python/hdfsutils.py

```diff
@@ -0,0 +1,79 @@
+import magic
+from hdfs import Config
+import nibabel as nib
+from gzip import GzipFile
+from io import BytesIO
+
+class HDFSUtils:
+    """Helper class for hdfs-related operations"""
+    
+    def __init__(self):
+        self.client = Config().get_client()    
+
+
+    #checks if filepath begins with hdfs://
+    def is_hdfs_uri(self, uri):
+        if 'hdfs://' in uri[:8]:
+            return True
+        return False
+
+    #return the hdfs path. Temp solution
+    def hdfs_path(self, uri):
+        #if uri is not an hdfs uri, return the uri provided
+        if not self.is_hdfs_uri(uri): return uri
+
+        if self.client.root is None:
+            #absolute path should begin at third slash since uris should be of format hdfs://host:port/path/to/file/or/folder
+            n = 3
+            path_split = uri.split('/')            
+            return '/'.join(path_split[n:])
+
+        return uri.split(self.client.root)[1]       
+
+    #returns the image loaded into nibabel
+    def get_slice(self, filepath, in_hdfs = None):
+        if in_hdfs is None:
+            in_hdfs = self.is_hdfs_uri(filepath)
+            if in_hdfs:
+                filepath = self.hdfs_path(filepath)
+            
+        if in_hdfs:
+            fh = None
+            with self.client.read(filepath) as reader:
+                stream = reader.read()
+                if self.is_gzipped(filepath, stream):
+                    fh = nib.FileHolder(fileobj=GzipFile(fileobj=BytesIO(stream)))
+                else:
+                    fh = nib.FileHolder(fileobj=BytesIO(stream))
+            
+            return nib.Nifti1Image.from_file_map({'header':fh, 'image':fh})
+
+        return nib.load(filepath)
+            
+                        
+    #checks is file is gzipped
+    #by default a filepath is required, however this function will not work
+    #with a filepath if file is located on HDFS. Therefore, if file is located
+    #on HDFS, the byte stream will also need to be provided   
```

Comment:
Can the library live with only the first bytes of the stream? We can't afford reading the whole file just to get its type.